src/v/cloud_storage/remote_partition.cc

```diff
@@ -828,32 +828,43 @@ remote_partition::aborted_transactions(offset_range offsets) {
         }
     } else {
         // Target archive section of the log
-        auto cur_res = co_await _manifest_view->get_cursor(
-          manifest.get_archive_start_offset());
-        if (cur_res.has_failure()) {
-            vlog(
-              _ctxlog.error,
-              "Failed to traverse archive part of the log: {}",
-              cur_res.error());
-            throw std::runtime_error(fmt_with_ctx(
-              fmt::format, "Failed to get the cursor {}", cur_res.error()));
-        }
-        auto cursor = std::move(cur_res.value());
         std::deque<std::pair<segment_meta, remote_segment_path>>
           meta_to_materialize;
-        co_await ss::repeat([&meta_to_materialize, &cursor, offsets] {
-            const auto& manifest = cursor->manifest()->get();
-            for (auto it = manifest.segment_containing(offsets.begin);
-                 it != manifest.end();
-                 ++it) {
-                if (it->base_offset > offsets.end_rp) {
-                    break;
+        while (true) {
```

Comment:
The chance of this going on forever is slim, right? otherwise, i would use a counter approach

Reply:
Feeling a sense of déjà-vu, I'm pretty sure we already had this exchange of messages already a couple of times 🤔